sources/platform/actors/development/actor_definition/input_schema/specification.md

```diff
@@ -1,27 +1,35 @@
+description: Learn how to define and validate a schema for your Actor's input with code examples. Provide an autogenerated input UI for your Actor's users.
 slug: /actors/development/actor-definition/input-schema/specification/v1
 ---
 
 # Actor input schema specification
 
-**Learn how to define and easily validate a schema for your Actor's input with code examples. Provide an autogenerated input UI for your Actor's users.**
+**Learn how to define and validate a schema for your Actor's input with code examples. Provide an autogenerated input UI for your Actor's users.**
 
 ---
 
-An Actor's input schema defines the input that the Actor accepts and the UI components used for input in Apify Console. Using input schema, you can provide an easy-to-use UI for your Actor's users and also ensure that the input provided is valid.
+The input schema serves two primary purposes for an Actor:
 
-Input schema can be embedded as an object in the **.actor/actor.json** file under the `input` field. Alternatively, you can store a path to a JSON file in that field. If the `input` field or **.actor/actor.json** are omitted, the input schema from **.actor/INPUT_SCHEMA.json** is used. In the absence of that file, the **INPUT_SCHEMA.json** file stored in the Actor's root directory is used. The file's maximum size is 100 kB. If the input schema is provided, then input is always validated to fulfill the schema when an Actor is being started (via API or from Apify Console).
+- It defines the structure and validation rules for the input data the Actor accepts.
+- It determines the user interface components rendered in the Apify Console for configuring the Actor's input.
 
-> You can also use our [visual input schema editor](https://apify.github.io/input-schema-editor-react) to guide you through creation of the **INPUT_SCHEMA.json** file.
-<!--  -->
-> If you need to validate your input schemas, you can use the [`apify vis`](/cli/docs/reference#apify-vis-path) command in the Apify CLI.
+By defining an input schema, you can provide a user-friendly interface for configuring your Actor while ensuring that the input data supplied by users adheres to the specified requirements and constraints.
+
+You can specify input schema for an Actor in multiple ways. One approach embeds it as an object within the `.actor/actor.json` file under the `input` field. Alternatively, provide a path to a `JSON` file containing the input schema in the same `input` field. If you omit the `input` field and the `.actor/actor.json` file, the system will look for an `INPUT_SCHEMA.json` file in the `actor` directory. In the absence of that file, it will search for an `INPUT_SCHEMA.json` file in the Actor's root directory. The max allowed size for the input schema file is 100 kB. When you provide an input schema, the system will validate the input data passed to the Actor during execution (via the API or the Apify Console) against the specified schema to ensure compliance before starting the Actor.
```

Comment:
```suggestion
You can specify input schema for an Actor in multiple ways. One approach embeds it as an object within the `.actor/actor.json` file under the `input` field. Alternatively, provide a path to a `JSON` file containing the input schema in the same `input` field. If you omit the `input` field and the `.actor/actor.json` file, the system will look for an `INPUT_SCHEMA.json` file in the `.actor` directory. In the absence of that file, it will search for an `INPUT_SCHEMA.json` file in the Actor's root directory. The max allowed size for the input schema file is 100 kB. When you provide an input schema, the system will validate the input data passed to the Actor during execution (via the API or the Apify Console) against the specified schema to ensure compliance before starting the Actor.
```